# 1. Data Quality Assessment

## Missing Values Check

## Duplicate Check

## Data Type Validation

## Descriptive Statistics

## Business Metrics Summary

## Initial Findings

In [23]:
import pandas as pd

df = pd.read_csv("../data/ecommerce_sales_data.csv")

#Missing Value
df.isnull().sum()

Order Date      0
Product Name    0
Category        0
Region          0
Quantity        0
Sales           0
Profit          0
dtype: int64

In [24]:
#Duplicate
df.duplicated().sum()

np.int64(0)

In [25]:
#Data Type
df.dtypes

Order Date       object
Product Name     object
Category         object
Region           object
Quantity          int64
Sales             int64
Profit          float64
dtype: object

In [26]:
df['Order Date'] = pd.to_datetime(df['Order Date'])

In [27]:
# Descriptive Statistic
df.describe().round(2)

,Order Date,Quantity,Sales,Profit
count,3500,3500.00,3500.00,3500.00
mean,2023-07-13 00:53:53.828571648,4.93,3047.97,527.05
min,2022-01-01 00:00:00,1.00,51.00,6.97
25%,2022-10-16 18:00:00,3.00,1049.50,158.70
50%,2023-07-21 12:00:00,5.00,2350.50,361.07
75%,2024-04-16 06:00:00,7.00,4537.00,729.12
max,2024-12-31 00:00:00,9.00,10782.00,2946.93
std,NaN,2.58,2440.21,504.14


In [28]:
# Business Metrics Summary
print(f"Total Sales   : {df['Sales'].sum():,.2f}")
print(f"Total Profit  : {df['Profit'].sum():,.2f}")
print(f"Total Quantity: {df['Quantity'].sum():,.0f}")

Total Sales   : 10,667,881.00
Total Profit  : 1,844,665.21
Total Quantity: 17,261


In [29]:
# Loss Transactions
loss_transactions = (df['Profit'] < 0).sum()

print(f"Jumlah transaksi rugi: {loss_transactions}")
print(f"Persentase transaksi rugi: {(loss_transactions/len(df))*100:.2f}%")

Jumlah transaksi rugi: 0
Persentase transaksi rugi: 0.00%


In [30]:
#Profit Margin
profit_margin = (df['Profit'].sum() / df['Sales'].sum()) * 100

print(f"Profit Margin: {profit_margin:.2f}%")

Profit Margin: 17.29%


In [31]:
summary = pd.DataFrame({
    'Metric': ['Total Rows',
               'Total Sales',
               'Total Profit',
               'Total Quantity',
               'Unique Products',
               'Unique Categories',
               'Unique Regions'],
    'Value': [
        len(df),
        df['Sales'].sum(),
        df['Profit'].sum(),
        df['Quantity'].sum(),
        df['Product Name'].nunique(),
        df['Category'].nunique(),
        df['Region'].nunique()
    ]
})

summary

,Metric,Value
0,Total Rows,3500.00
1,Total Sales,10667881.00
2,Total Profit,1844665.21
3,Total Quantity,17261.00
4,Unique Products,10.00
5,Unique Categories,3.00
6,Unique Regions,4.00


## Initial Findings
Based on the initial data quality assessment and descriptive statistics, several key observations were identified:

- The dataset contains 3,500 transaction records.
- No missing values or duplicate records were found, indicating good data quality for further analysis.
- Total sales reached 10.67 million, generating a total profit of 1.84 million.
- A total of 17,261 units were sold during the observed period.
- The business operates across 4 regions and offers products from 3 categories.
- There are 10 unique products available in the dataset.
- The profit-to-sales ratio is approximately 17.29%, suggesting that the business generates positive profitability overall.
- Further analysis is required to identify which products, categories, and regions contribute the most to revenue and profit.